[feat] 커피빵에서 빵만 남기기

theminjunchoi opened the pull request:
# ✅ 체크리스트

- [x] merge 타겟 브랜치 잘 설정되었는지 확인하기 (fe/dev, be/dev)

# 🔥 연관 이슈

- close #1004 

# 🚀 작업 내용

1. 프로덕션과 테스트에서 메뉴관련된 부분 전부 수정했습니다.
2. 아직 메뉴 관련 코드를 "삭제"하진 않았어요. 의견주세요!


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * 룸 생성·입장 흐름에서 메뉴 선택 관련 입력이 제거되어 API와 사용자 흐름이 단순해졌습니다
  * 플레이어 및 룸 내 메뉴 선택/표시 흐름이 정리되고 관련 로그가 간소화되었습니다
* **Removed**
  * 메뉴 관련 공개 API, 도메인 타입, 초기화용 데이터/컴포넌트 및 연관된 테스트·컨트롤러가 제거되었습니다

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

theminjunchoi committed:
refactor: Room 생성 및 입장 로직에서 메뉴 관련 코드 제거

- RoomCreateEvent 및 RoomJoinEvent의 생성자에서 SelectedMenuRequest 제거
- Room 관련 도메인 및 서비스 계층의 메뉴 선택 로직 및 필드 제거
- Player의 selectedMenu 필드와 관련 메서드 삭제
- RoomRestController의 createRoom 및 enterRoom 메서드에서 메뉴 관련 파라미터 제거
- 관련 테스트 및 기타 종속 로직 수정 및 리팩토링
- 코드 간결성 및 비즈니스 로직 일치성 향상

theminjunchoi committed:
refactor: 테스트 코드에서 메뉴 관련 로직 제거

- Room 및 Player의 메뉴 처리 로직에 대한 테스트 코드 삭제
- RoomCreateEventHandlerTest, RoomCommandServiceTest 등에서 SelectedMenuRequest 제거
- Room 관련 기타 테스트 및 Fixture 클래스에서 불필요한 메뉴 관련 종속성 제거
- 필요 없는 의존성과 인수 제거로 테스트 로직 간결화

20HyeonsuLee reviewed (changes requested):
우선, 번거로운 작업 맡아주셔서 감사해요 🙇‍♂️🙇‍♂️

메뉴 관련된 코드는 전부 삭제해도 좋을 것 같아요!

1. 하위호환성을 위해 메뉴 관련 api, domain을 모두 살려둔다.
2. 하위호환성 상관 없으니 모두 지운다.

요렇게 두 방식이 있을 것 같은데, 점검중 띄워놓기도 했고 아차피 프론트랑 같이 배포할거니 2번 방식이 좋다고 생각합니다!

우선 RC로 남겨두겠습니다!

junhaa reviewed (approved):
저도 한스랑 같은 의견이라서, 제거해도 될 것 같아요

우선 어푸는 해두겠습니다!

kiwoook commented on backend/src/main/java/coffeeshout/room/ui/request/RoomCreateRequest.java:
```diff
@@ -1,7 +1,6 @@
-        String hostName,
-        Long menuId
+        String hostName
 ) {
```
추가해놓으면 좋을 듯 합니다

theminjunchoi committed:
refactor: RoomService 및 LogAspect에서 메뉴 관련 로직 제거

- LogAspect의 logEnterRoom 및 logSelectMenu 메서드에서 메뉴 필드와 로깅 제거
- SelectedMenu 및 SelectedMenuRequest에 대한 도메인 및 종속성 제거
- 코드 간결화 및 메뉴 관련 불필요한 로직 삭제

theminjunchoi committed:
refactor: 메뉴 관련 도메인 및 서비스 계층 삭제

- 메뉴와 관련된 모든 엔티티, 서비스, 초기화 로직 및 API 삭제
- MemoryMenuRepository와 MemoryMenuCategoryRepository 등 모든 메뉴 종속 리포지토리 제거
- 메뉴와 관련된 DTO, 엔드포인트, 테스트 코드 및 설정 클래스 정리
- 코드 간결성 및 유지보수 용이성 향상

theminjunchoi committed:
test: 메뉴 관련 모든 테스트 코드 삭제

- 메뉴와 관련된 모든 테스트 클래스 제거
- CustomMenuTest, MenuCategoryFixture 등 테스트 관련 Fixture 삭제
- 테스트 간 의존성 감소 및 코드 간결화

20HyeonsuLee reviewed (approved):
빵만 남기느라 고생하셨어요👍👍

`resources/data`에 있는 `menu-category-data.yml`, `menu-data.yml`도 삭제해도 될 것 같습니다. 어프로브 해둘테니 삭제하고 바로 머지 해주시죠!

theminjunchoi committed:
refactor: 메뉴 데이터 YAML 파일 삭제

- menu-category-data.yml 및 menu-data.yml 파일 제거
- 메뉴 관련 모든 데이터 정리로 코드 간소화 및 무결성 유지
